### DISCUSSION TITLE
Support for Electron 14 (nodeIntegration, contextIsolation switch)

### DISCUSSION THREAD

**Discussion by famibee** [pixijs/pixijs#7794] [Q&A] [answered]
I would like to request support for Electron 14. (NodeIntegration, contextIsolation switch)
> Electron 14 への対応をお願いしたいです。（nodeIntegration、contextIsolation スイッチ）

If nodeIntegration is set to false or contextIsolation is set to true, [Reference Error: require is not defined] will occur.
> nodeIntegrationを false にしたり、contextIsolation を true にすると【ReferenceError: require is not defined】になってしまう不具合です。

Since it is a mandatory setting from Electron 14, please support it.
> Electron 14 からは必須の設定となりますので、対応をお願いいたします。

I will attach a small sample project, so please open it with VS Code.
> 小さなサンプルプロジェクトを添付しますので、VSCodeにて開いてみて下さい。

### Expected Behavior
> fn:app4webpack.js 🎃
> fn:SysApp.ts 🍭 tx:Texture
<img width="221" alt="2021-09-14 09-14 火 21 09" src="https://user-images.githubusercontent.com/16816132/133255442-f580ddd4-4a04-4e45-a862-f7e4417fcd3d.png">

### Current Behavior
- [Reference Error: require is not defined]
<img width="518" alt="2021-09-14 09-14 火 21 10" src="https://user-images.githubusercontent.com/16816132/133255297-47d0f33a-372c-4808-a2d0-15db935a1aee.png">

### Steps to Reproduce
- Unzip the [electron14_tst.zip](https://github.com/pixijs/pixijs/files/7161679/electron14_tst.zip)

- npm i
- task "webpack:dev:w"
- (Change comment out to switch doc/app.js "const wp = {}" )
- task "start"
<img width="530" alt="20210914el14" src="https://user-images.githubusercontent.com/16816132/133254809-cafaac31-dbfa-4c52-b4a5-616284a749a3.png">

### Environment

- **`pixi.js` version**: 6.1.3
- **Electron & Version**: 13.4.0
- **OS & Version**: macOS Big Sur 11.6
- **Running Example**: [electron14_tst.zip](https://github.com/pixijs/pixijs/files/7161679/electron14_tst.zip)

**Comment by bigtimebuddy** [comment, marked as answer]
If you're using `contextIsolation` true and `nodeIntegration` false, then you need your target to be `web` instead of `electron-renderer` (core/webpack.config.js). Changing this worked. Here's what the webpack docs say about `electron-renderer`:

> Compile for Electron for renderer process, providing a target using JsonpTemplatePlugin, FunctionModulePlugin for browser environments and NodeTargetPlugin and ExternalsPlugin for CommonJS and Electron built-in modules.

So yeah, this will not work with nodeIntegration false.

_Also, for help with third-party tools like webpack or electron, please create a discussion. Issues are for specific bug in the code, not for support._

**Reply by famibee** [reply]
I didn't study, thank you for your professor!
> 不勉強でした、ご教授ありがとうございます！
